substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/AbstractDynamicClassGenerationSupport.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2019, 2019, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020 Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent;
+
+import com.oracle.svm.core.util.JavaClassUtil;
+import static com.oracle.svm.jvmtiagentbase.Support.jniFunctions;
+import static com.oracle.svm.jvmtiagentbase.Support.getMethodFullNameAtFrame;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.oracle.svm.jni.nativeapi.JNIMethodId;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+import org.graalvm.nativeimage.c.type.VoidPointer;
+import org.graalvm.word.WordFactory;
+
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
+
+public abstract class AbstractDynamicClassGenerationSupport {
+
+    protected JNIEnvironment jni;
+    protected JNIObjectHandle callerClass;
+    protected final String generatedClassName;
+    protected TraceWriter traceWriter;
+    protected NativeImageAgent agent;
+    protected int generatedClassHashCode = 0;
+    protected byte[] values = null;
+
+    private static Path dynclassDumpDir = null;
+    private static final Path DEFAULT_DUMP = Paths.get("dynClass");
```

Comment:
I think it would be better to keep the dynamic classes in the sam folder as other config. This way it will be easier to use native image with `-H:ConfigurationFileDirectories=`. This is, for example the case in our benchmarks.

Reply:
Agreed, as discussed previously (https://github.com/oracle/graal/pull/2442#issuecomment-676704131 and https://github.com/oracle/graal/pull/2442#issuecomment-677381880). I'll trace all dynamic generated classes in a new configuration file and store all generated classes under the same directory.

Reply:
Thank you for your comments @vjovanov :+1: 
@ziyilin , please do address https://github.com/oracle/graal/pull/2442#issuecomment-676704131 and https://github.com/oracle/graal/pull/2442#issuecomment-677381880 and also consider which suggestions from the object serialization PR might apply here as well. Then I will review in more depth.